lib/honeycomb/integrations/rack.rb

```diff
@@ -33,15 +36,25 @@ def call(env)
           span.add_field(key, value)
         end
 
+        is_secure = lambda do
+          env["rack.url_scheme"] == "https" || env["rack.url_scheme"] == "wss"
+        end
+
+        is_xhr = -> { env["HTTP_X_REQUESTED_WITH"] == "XMLHttpRequest" }
+
         extract_fields(env, RACK_FIELDS, &add_field)
 
+        span.add_field("request.secure", is_secure.call)
+        span.add_field("request.xhr", is_xhr.call)
```

Comment:
why the procs/lambdas here over just variables?

Reply:
@martin308 I went back and forth on that :) The short version is that the lambdas make the code a bit shorter and in the case of `is_secure` at least, encapsulates some business logic (the environment value can be x or y).

Reply:
I don't think I have strong feelings either way, it just stood out as being a bit different. Is there any performance implication to doing it this way instead of just assigning them to variables?

Reply:
@martin308 I replaced with just straight forward assignment anyway. Sat with it and don't think it justifies the difference.

Reply:
ok cool. I did also just look up the rack docs and potentially we could just rely on the rack methods for this and maybe some of the other bits in here. I don't remember if I considered that when adding these fields originally. There is [xhr?](https://www.rubydoc.info/gems/rack/Rack%2FRequest%2FHelpers:xhr%3F) and [ssl?](https://www.rubydoc.info/gems/rack/Rack%2FRequest%2FHelpers:ssl%3F)